File: QolmsYappli/QolmsYappli/AzureStorage/AccessTable/Writer/AccessTableEntityWriter.vb
Imports MGF.QOLMS.QolmsAzureStorageCoreV1

''' <summary>
''' アクセス ログ テーブル ストレージ へ値を登録するための機能を提供します。
''' この クラス は継承できません。
''' </summary>
''' <typeparam name="TTableEntity">アクセス ログ テーブル ストレージ エンティティ の型。</typeparam>
''' <remarks></remarks>
Friend NotInheritable Class AccessTableEntityWriter(Of TTableEntity As {QsAccessTableEntityBase, New})
    Inherits QsAzureTableStorageWriterBase(Of TTableEntity)
    Implements IQsAzureTableStorageWriter(Of TTableEntity, AccessTableEntityWriterArgs(Of TTableEntity), AccessTableEntityWriterResults)

#Region "Constructor"

    ''' <summary>
    ''' <see cref="AccessTableEntityWriter(Of TTableEntity)" /> クラス の新しい インスタンス を初期化します。
    ''' </summary>
    ''' <remarks></remarks>
    Public Sub New()

        MyBase.New(True)

    End Sub

#End Region

#Region "IQsAzureTableStorageWriter Support"

    ''' <summary>
    ''' Azure テーブル ストレージ へ値を設定します。
    ''' </summary>
    ''' <param name="args">引数 クラス。</param>
    ''' <returns>
    ''' 戻り値 クラス。
    ''' </returns>
    ''' <remarks></remarks>
    Public Function Execute(args As AccessTableEntityWriterArgs(Of TTableEntity)) As AccessTableEntityWriterResults Implements IQsAzureTableStorageWriter(Of TTableEntity, AccessTableEntityWriterArgs(Of TTableEntity), AccessTableEntityWriterResults).Execute

        Dim result As New AccessTableEntityWriterResults() With {.IsSuccess = False}
        Dim entity As New TTableEntity()

        With entity
            .SetPartitionKey(args.AccessDate)
            .SetRowKey(args.AccessDate, args.AccountKey)
            .AccessType = args.AccessType
            .AccessUri = args.AccessUri
            .Comment = args.Comment
            .UserHostAddress = args.UserHostAddress
            .UserHostName = args.UserHostName
            .UserAgent = args.UserAgent
        End With

        result.IsSuccess = Me.InsertOrUpdateEntities(New List(Of TTableEntity)() From {entity}).Count = 1
        result.Result = If(result.IsSuccess, 1, 0)

        Return result

    End Function

#End Region

End Class

File: QolmsJotoWebView/QolmsJotoWebView/AzureStorage/AccessTable/Writer/AccessTableEntityWriter.cs
using MGF.QOLMS.QolmsAzureStorageCoreV1;
using System.Collections.Generic;

namespace MGF.QOLMS.QolmsJotoWebView 
{ 
    /// <summary>
    /// アクセス ログ テーブル ストレージ へ値を登録するための機能を提供します。
    /// この クラス は継承できません。
    /// </summary>
    /// <typeparam name="TTableEntity">アクセス ログ テーブル ストレージ エンティティ の型。</typeparam>
    internal sealed class AccessTableEntityWriter<TTableEntity>
        : QsAzureTableStorageWriterBase<TTableEntity>, IQsAzureTableStorageWriter<TTableEntity, AccessTableEntityWriterArgs<TTableEntity>, AccessTableEntityWriterResults> where TTableEntity : QsAccessTableEntityBase, new()
    {

        #region "Constructor"

        /// <summary>
        /// <see cref="AccessTableEntityWriter" /> クラス の新しい インスタンス を初期化します。
        /// </summary>
        public AccessTableEntityWriter() : base(true) { }

        #endregion

        #region "IQsAzureTableStorageWriter Support"

        /// <summary>
        /// Azure テーブル ストレージ へ値を設定します。
        /// </summary>
        /// <param name="args">引数 クラス。</param>
        /// <returns>
        /// 戻り値 クラス。
        /// </returns>
        AccessTableEntityWriterResults IQsAzureTableStorageWriter<TTableEntity, AccessTableEntityWriterArgs<TTableEntity>, AccessTableEntityWriterResults>.Execute(AccessTableEntityWriterArgs<TTableEntity> args) 
        {
            var result = new AccessTableEntityWriterResults() { IsSuccess = false };
            var entity = new TTableEntity();

            entity.SetPartitionKey(args.AccessDate);
            entity.SetRowKey(args.AccessDate, args.AccountKey);
            entity.AccessType = args.AccessType;
            entity.AccessUri = args.AccessUri;
            entity.Comment = args.Comment;
            entity.UserHostAddress = args.UserHostAddress;
            entity.UserHostName = args.UserHostName;
            entity.UserAgent = args.UserAgent;

            result.IsSuccess = this.InsertOrUpdateEntities(new List<TTableEntity>() { entity }).Count == 1;
            result.Result = result.IsSuccess ? 1 : 0;

            return result;
        }

        #endregion

    }
}